src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMCheckoutListener.java

```diff
@@ -0,0 +1,9 @@
+package com.atlassian.bitbucket.jenkins.internal.scm;
+
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+public interface BitbucketSCMCheckoutListener {
```

Comment:
I created this class so that we can separate the build status posting and the deployment posting

Reply:
/nit Not directly related to this PR but I notice we switch between SCM (all caps) and Scm. Is there a reason for this or is it just an accidental inconsistency that's crept it?

Reply:
Accidental inconsistency, we should be consistent all caps.